File: step_definitions/Multiple Channels to Contact Customer Service.js
describe('Multiple Channels to Contact Customer Service', () => {
  beforeEach(() => {
    cy.visit('/customer-service');
  });

  it('should display the customer service page or contact information', () => {
    cy.get('.customer-service-page').should('be.visible');
  });

  it('should display phone contact information', () => {
    cy.get('.phone-contact').should('be.visible').and('contain', 'Phone Number').and('contain', 'Working Hours');
  });

  it('should display email contact information', () => {
    cy.get('.email-contact').should('be.visible').and('contain', 'Email Address').and('contain', 'Response Time');
  });

  it('should have a live chat option available', () => {
    cy.get('.live-chat-option').should('be.visible');
  });

  it('should open phone dialer with phone number pre-filled', () => {
    cy.get('.phone-contact').click();
    cy.get('.phone-dialer').should('be.visible').and('contain', 'Phone Number');
  });

  it('should open email client with recipient email address pre-filled', () => {
    cy.get('.email-contact').click();
    cy.get('.email-client').should('be.visible').and('contain', 'Email Address');
  });

  it('should open live chat window', () => {
    cy.get('.live-chat-option').click();
    cy.get('.live-chat-window').should('be.visible');
  });

  it('should send message in live chat and receive response', () => {
    cy.get('.live-chat-window').type('Hello, I need help.');
    cy.get('.send-message-button').click();
    cy.get('.chat-response').should('be.visible').and('contain', 'Response');
  });

  it('should display alternative contact methods if available', () => {
    cy.get('.alternative-contact-methods').should('be.visible');
  });

  it('should mention expected response time for each contact method', () => {
    cy.get('.response-time').should('be.visible');
  });

  it('should mention working hours for each contact method', () => {
    cy.get('.working-hours').should('be.visible');
  });

  it('should be reachable through each available channel', () => {
    cy.get('.contact-channel').each(($channel) => {
      cy.wrap($channel).click();
      cy.get('.channel-response').should('be.visible');
    });
  });

  it('should have contact information easily accessible from any page', () => {
    cy.get('.global-contact-info').should('be.visible');
  });

  it('should display contact information correctly on different devices', () => {
    cy.viewport('iphone-x');
    cy.get('.customer-service-page').should('be.visible');
    cy.viewport('macbook-15');
    cy.get('.customer-service-page').should('be.visible');
  });

  it('should have up to date and accurate contact information', () => {
    cy.get('.contact-info').should('be.visible').and('contain', 'Updated Info');
  });
});

File: src/test/java/com/cucumber/steps/Multiple Channels to Contact Customer Service.java
package com.cucumber.steps;

import com.framework.cucumber.TestHarness;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;

public class CustomerServiceSteps extends CustomerServicePage {

    private TestHarness testHarness = new TestHarness();

    @Given("I have accessed the system's customer service page or contact information")
    public void iHaveAccessedTheCustomerServicePage() {
        navigateToCustomerServicePage();
    }

    @Then("the customer service page or contact information should be displayed")
    public void customerServicePageShouldBeDisplayed() {
        verifyCustomerServicePageDisplayed();
    }

    @Then("phone contact information should be displayed, including phone number and working hours")
    public void phoneContactInformationShouldBeDisplayed() {
        verifyPhoneContactInformation();
    }

    @Then("email contact information should be displayed, including email address and response time")
    public void emailContactInformationShouldBeDisplayed() {
        verifyEmailContactInformation();
    }

    @Then("a live chat option should be available for customers to contact customer service")
    public void liveChatOptionShouldBeAvailable() {
        verifyLiveChatOptionAvailable();
    }

    @When("I click on the phone contact information")
    public void iClickOnPhoneContactInformation() {
        clickPhoneContactInformation();
    }

    @Then("the phone dialer should open with the provided phone number pre-filled")
    public void phoneDialerShouldOpen() {
        verifyPhoneDialerOpened();
    }

    @When("I click on the email contact information")
    public void iClickOnEmailContactInformation() {
        clickEmailContactInformation();
    }

    @Then("the default email client should open with the recipient email address pre-filled")
    public void emailClientShouldOpen() {
        verifyEmailClientOpened();
    }

    @When("I click on the live chat option")
    public void iClickOnLiveChatOption() {
        clickLiveChatOption();
    }

    @Then("a live chat window should open for customers to start a conversation with customer service")
    public void liveChatWindowShouldOpen() {
        verifyLiveChatWindowOpened();
    }

    @Given("I have opened the live chat window")
    public void iHaveOpenedLiveChatWindow() {
        openLiveChatWindow();
    }

    @When("I enter a message in the live chat window")
    public void iEnterMessageInLiveChatWindow() {
        enterMessageInLiveChatWindow();
    }

    @Then("the message should be sent to customer service and a response should be received")
    public void messageShouldBeSentAndResponseReceived() {
        verifyMessageSentAndResponseReceived();
    }

    @Then("alternative contact methods should be displayed, if available, with relevant information")
    public void alternativeContactMethodsShouldBeDisplayed() {
        verifyAlternativeContactMethods();
    }

    @Then("the expected response time for each contact method should be mentioned")
    public void responseTimeShouldBeMentioned() {
        verifyResponseTimeMentioned();
    }

    @Then("the working hours for each contact method should be mentioned")
    public void workingHoursShouldBeMentioned() {
        verifyWorkingHoursMentioned();
    }

    @When("I try contacting customer service using each available channel")
    public void tryContactingCustomerService() {
        contactCustomerServiceThroughAllChannels();
    }

    @Then("customer service should be reachable through each available channel")
    public void customerServiceShouldBeReachable() {
        verifyCustomerServiceReachable();
    }

    @Then("the customer service contact information should be easily accessible from any page of the system")
    public void contactInformationShouldBeAccessible() {
        verifyContactInformationAccessible();
    }

    @Then("the customer service contact information should be displayed correctly on different devices without any formatting issues")
    public void contactInformationShouldBeDisplayedCorrectly() {
        verifyContactInformationDisplayedCorrectly();
    }

    @Then("the customer service contact information should be up to date and accurate")
    public void contactInformationShouldBeUpToDate() {
        verifyContactInformationUpToDate();
    }

    @Then("customer service contact forms or live chat should include measures to prevent spam or automated abuse, such as CAPTCHA")
    public void contactOptionsShouldBeProtectedFromSpam() {
        verifySpamProtectionMeasures();
    }

    @When("I attempt to print the customer service contact information page")
    public void attemptToPrintContactInformation() {
        initiatePrintContactInformation();
    }

    @Then("the contact information should be formatted correctly for printing")
    public void contactInformationShouldBePrintable() {
        verifyPrintFormatOfContactInformation();
    }
}